client/src/main/scala/drt/client/modules/GoogleAnalytics.scala

```diff
@@ -1,26 +1,82 @@
 package drt.client.modules
 
-
+import org.scalajs.dom
 import scala.scalajs.js
 import scala.scalajs.js.annotation.JSGlobalScope
 
 object GoogleEventTracker {
-  def isScriptLoaded: Boolean =js.Dynamic.global.analytics.isInstanceOf[js.Function]
-  def sendPageView(page:String):Unit={
-    if (isScriptLoaded) GoogleAnalytics.analytics("send","event","pageView", page)
+  val trackingCode: GoogleTrackingCode = GoogleTrackingCode.getTrackingCode(dom.document.location.href)
+  def port: String = dom.document.getElementById("port-code").getAttribute("value")
+  def userId: String = dom.document.getElementById("user-id").getAttribute("value")
+  def isScriptLoaded: Boolean = js.Dynamic.global.analytics.isInstanceOf[js.Function]
+  private var hasCreateTrackerRun = false
+
+  private def runCreateTracker(): Unit = {
+    if (!hasCreateTrackerRun && !userId.isEmpty && !port.isEmpty) {
+      GoogleAnalytics.analytics("create", trackingCode.code, "auto", js.Dictionary("userId"->userId))
+      hasCreateTrackerRun = true
+    }
+  }
+
+  def sendPageView(page: String): Unit = {
+    if (isScriptLoaded) {
+      runCreateTracker()
+      if (hasCreateTrackerRun) {
+        GoogleAnalytics.analytics("set", "page", s"/$port/$page")
+        GoogleAnalytics.analytics("send", "pageView")
+      }
+    }
   }
-  def sendEvent(category:String,action:String,label:String):Unit={
-    if (isScriptLoaded) GoogleAnalytics.analytics("send","event",category,action,label)
+
+  def sendEvent(category: String, action: String, label: String): Unit = {
+    if (isScriptLoaded && hasCreateTrackerRun) GoogleAnalytics.analytics("send", "event", s"$port-$category", action, label)
   }
-  def sendEvent(category:String,action:String,label:String,value:String):Unit={
-    if (isScriptLoaded) GoogleAnalytics.analytics("send","event",category,action,label,value)
+
+  def sendEvent(category: String, action: String, label: String, value: String): Unit = {
+    if (isScriptLoaded && hasCreateTrackerRun) GoogleAnalytics.analytics("send", "event", s"$port-$category", action, label, value)
+  }
+
+  def sendError(description: String, fatal: Boolean): Unit = {
+    if (isScriptLoaded && hasCreateTrackerRun) GoogleAnalytics.analytics("send", "exception", js.Dictionary("exDescription" -> description, "exFatal" -> fatal))
+  }
+}
+
+trait GoogleTrackingCode {
+  val code: String
+}
+
+case object ProductionTrackingCode extends GoogleTrackingCode {
+  val code = "UA-125317287-1"
```

Comment:
should we rather make these environment variables?

Reply:
I couldn't figure out if we can read a config property from the properties file. 

I didn't want to go to the backend to ask what the property - get the response - put it in a div somewhere in the page and pull it out.

There are only ever going to be two values one for production and another that isn't. The client can dictate from the url which one it should use.

Reply:
The URL is likely to change, and is a bit of a hacky way to tell the app which environment it's in. You can set it in the template:
https://github.com/UKHomeOffice/drt-v2/blob/master/server/src/main/twirl/views/index.scala.html

https://github.com/UKHomeOffice/drt-v2/blob/master/server/src/main/scala/controllers/Application.scala#L383

so avoid requiring an http request to set it.

Reply:
to be honest, we probably don't even need tracking on notprod, so might be easier to just turn it off completely with an environment variable.

Reply:
I've changed it so that it doesn't track if the environment variable isn't there